.github/workflows/deploy_docs.yml

```diff
@@ -0,0 +1,38 @@
+name: Deploy Docs to GitHub Pages
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'docs/**'
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+          cache: yarn
+
+      - name: Change to docs directory
+        run: cd docs
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Build the Docusaurus site
+        run: yarn build
+
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          deploy_key: ${{ secrets.DOCS_DEPLOY_KEY }}
+          external_repository: 'bagetter/bagetter.github.io'
+          publish_dir: ./docs/build
+          publish_branch: main
```

Comment:
maybe add some extra information like the user or commit message like the following lines:
```
user_name: 'github-actions[bot]'
user_email: 'github-actions[bot]@users.noreply.github.com'
commit_message: ${{ github.event.head_commit.message }}
```